perllib/FixMyStreet/DB/Result/Problem.pm

```diff
@@ -660,16 +660,21 @@ sub body {
         }
     } else {
         my $bodies = $problem->bodies;
-        $body = join( _(' and '),
-            map {
-                my $name = $_->name;
-                if ($c and FixMyStreet->config('AREA_LINKS_FROM_PROBLEMS')) {
-                    '<a href="' . $_->url($c) . '">' . $name . '</a>';
-                } else {
-                    $name;
-                }
-            } values %$bodies
-        );
+        my @body_names = sort map {
+            my $name = $_->name;
+            if ($c and FixMyStreet->config('AREA_LINKS_FROM_PROBLEMS')) {
+                '<a href="' . $_->url($c) . '">' . $name . '</a>';
+            } else {
+                $name;
+            }
+        } values %$bodies;
+        if ( scalar @body_names > 2 ) {
+            my $l = scalar @body_names;
+            $body = join( _(', ', ), @body_names[0..$l-2]);
+            $body = join( _(', and '), ($body, $body_names[$l-1]));
+        } else {
```

Comment:
`splice @body_names, 0, -1` and `$body_names[-1]` might be a more perly way to do this.